singularity-eos/eos/eos_vinet.hpp

```diff
@@ -0,0 +1,383 @@
+  PORTABLE_INLINE_FUNCTION Real TExpansionCoeffFromDensityTemperature(
+      const Real rho, const Real temp, Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION Real GruneisenParamFromDensityTemperature(
+      const Real rho, const Real temp, Real *lambda = nullptr) const {
+    return robust::ratio(_A0 * _B0, _Cv0 * rho);
+  }
+  PORTABLE_INLINE_FUNCTION Real GruneisenParamFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    return robust::ratio(_A0 * _B0, _Cv0 * rho);
+  }
+  PORTABLE_INLINE_FUNCTION void FillEos(Real &rho, Real &temp, Real &energy, Real &press,
+                                        Real &cv, Real &bmod, const unsigned long output,
+                                        Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                              Real &bmod, Real &dpde, Real &dvdt,
+                              Real *lambda = nullptr) const;
+  // Generic functions provided by the base class. These contain e.g. the vector
+  // overloads that use the scalar versions declared here
+  SG_ADD_BASE_CLASS_USINGS(Vinet)
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return 0; }
+  static constexpr unsigned long PreferredInput() { return _preferred_input; }
+  PORTABLE_INLINE_FUNCTION void PrintParams() const {
+    static constexpr char st[]{"Vinet Params: "};
+    printf("%s rho0:%e T0:%e B0:%e BP0:%e\n  A0:%e Cv0:%e E0:%e S0:%e\n"
+           "non-zero elements in d2tod40 array:\n",
+           st, _rho0, _T0, _B0, _BP0, _A0, _Cv0, _E0, _S0);
+    for (int i = 0; i < 39; i++) {
+      if (_d2tod40[i] > 0.0) printf("d%i:%e\t", i + 2, _d2tod40[i]);
+    }
+    printf("\n\n");
+  }
+  // Density/Energy from P/T not unique, if used will give error
+  PORTABLE_INLINE_FUNCTION void
+  DensityEnergyFromPressureTemperature(const Real press, const Real temp, Real *lambda,
+                                       Real &rho, Real &sie) const;
+  inline void Finalize() {}
+  static std::string EosType() { return std::string("Vinet"); }
+  static std::string EosPyType() { return EosType(); }
+
+ private:
+  static constexpr const unsigned long _preferred_input =
+      thermalqs::density | thermalqs::temperature;
+  Real _rho0, _T0, _B0, _BP0, _A0, _Cv0, _E0, _S0;
+  static constexpr const int PressureCoeffsd2tod40Size = 39;
+  static constexpr const int VinetInternalParametersSize = PressureCoeffsd2tod40Size + 4;
+  Real _VIP[VinetInternalParametersSize], _d2tod40[PressureCoeffsd2tod40Size];
+  void CheckVinet();
+  void InitializeVinet(const Real *expcoeffs);
+  PORTABLE_INLINE_FUNCTION void Vinet_F_DT_func(const Real rho, const Real T,
+                                                Real *output) const;
+};
+
+PORTABLE_INLINE_FUNCTION void Vinet::CheckVinet() {
+
+  if (_rho0 < 0.0) {
+    printf("testing rho0\n");
```

Comment:
This print statement and the others like it that are before a `PORTABLE_ALWAYS_THROW_OR_ABORT` should be removed now that we know the throw machinery is working. That way the code will be quieter during normal operation.